java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -133,41 +136,35 @@ private void update(Symbol m) {
     private List<List<Type>> computeChains(Symbol m, Type type) {
       Symbol.TypeSymbol typeSymbol = type.symbol();
       Set<ClassJavaType> superTypes = ((JavaSymbol.TypeJavaSymbol) typeSymbol).directSuperTypes();
-      List<List<Type>> result = new ArrayList<>();
-      for (Type superType: superTypes) {
-        for (List<Type> chain: computeChains(m, superType)) {
-          List<Type> newChain = new ArrayList<>();
-          newChain.addAll(chain);
-          newChain.add(type);
-          result.add(newChain);
-        }
-      }
-      boolean definesSymbol = definesSymbol(m, typeSymbol, false);
+      List<List<Type>> result = superTypes.stream()
+        .flatMap(superType -> computeChains(m, superType).stream())
+        .map(c -> Stream.concat(c.stream(), Stream.of(type)).collect(Collectors.toList()))
```

Comment:
adding to the end of the list should be sufficient rather than going from Stream + collect